### PR TITLE
Add SHOW CREATE FUNCTION

### PR DESCRIPTION
**Pull request by electrum**
## Description

Add a new SQL statement `SHOW CREATE FUNCTION`.

This PR also cleans up some things in the parser.

Commit `Extract functions for ShowCreate types` should be reviewed ignoring whitespace.

## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# General
* Add {doc}`/sql/show-create-function`. ({issue}`21809`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/security/DenyAllAccessControl.java [hunk 1]
```diff
@@ -568,4 +568,10 @@ public void checkCanDropFunction(SecurityContext context, QualifiedObjectName fu
     {
         denyDropFunction(functionName.toString());
     }
+
+    @Override
+    public void checkCanShowCreateFunction(SecurityContext context, QualifiedObjectName functionName)
+    {
+        denyShowCreateTable(functionName.toString());
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/security/DenyAllAccessControl.java:575]: shouldn't that be `denyShowCreateFunction`?

File: core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java [hunk 2]
```diff
@@ -713,6 +722,26 @@ private Query showCreateSchema(ShowCreate node)
...
 
+        private Node showCreateFunction(ShowCreate node)
+        {
+            QualifiedObjectName functionName = qualifiedFunctionName(functionSchema, node, node.getName());
+
+            accessControl.checkCanShowCreateFunction(session.toSecurityContext(), functionName);
+
+            Collection<LanguageFunction> functions = metadata.getLanguageFunctions(session, functionName);
+            if (functions.isEmpty()) {
+                throw semanticException(NOT_FOUND, node, "Function not found");
```
Comment (wendigo) [inline core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java:733]: nit: `"Function %s not found".formatted(functionName)`

**Review by wendigo — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java (hunk 2 above)
Comment (electrum) [reply core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java:733]: This matches what we do for other statements like `DROP FUNCTION`. This statement is for a specific function, so having the name doesn't seem to provide any value, and I think it's better to have a constant (non-variable) error message.

**Review by martint — APPROVED** [review]

**Comment by mosabua** [comment]
Just saw this in the release notes! So awesome. We might need a bit more docs ... I will look.

**Pull request merged**